sceptre/plan/actions.py

```diff
@@ -1000,3 +1000,115 @@ def diff(self, stack_differ):
         :rtype: sceptre.diffing.stack_differ.StackDiff
         """
         return stack_differ.diff(self)
+
+    @add_stack_hooks
+    def detect_stack_drift(self):
+        """
+        Detects stack drift for a running stack.
+
+        :returns: The stack drift.
+        :rtype: Tuple[str, Union[str, dict]]
+        """
+        response = self._detect_stack_drift()
+
+        detection_id = response["StackDriftDetectionId"]
+        status = self._wait_for_drift(detection_id)
+
+        if status == "DETECTION_COMPLETE":
+            response = self._describe_stack_resource_drifts()
+            return_value = (self.stack.external_name, response)
+        else:
+            return_value = (self.stack.external_name, status)
```

Comment:
Why would you not raise an error if drift detection did not succeed or if it is still in progress?

Reply:
I don't think that an error in drift detection should result in an error in Sceptre. Sceptre is just the messenger here, reporting back on how things are in drift detection.

Reply:
If drift detection fails, I absolutely DO think an error ought to be raised. Otherwise, this command will proceed as if there is no problem.

Reply:
@jfalkenstein Yes, I see your point here. I was not aware that the external name is already available somehow in the CLI. I could fix this up in the way that you propose.

Reply:
How about now?